src/extensions/renderer/canvas/drawing-edges.js

```diff
@@ -208,8 +208,18 @@ CRp.drawEdgePath = function( edge, context, pts, type ){
       case 'straight':
       case 'segments':
       case 'haystack':
-        for( let i = 2; i + 1 < pts.length; i += 2 ){
-          context.lineTo( pts[ i ], pts[ i + 1] );
+        const taxiRadius= edge.pstyle( 'taxi-radius' )
+        const taxiRadiusValue = taxiRadius ? taxiRadius.pfValue : 0
```

Comment:
Similar to how `border-radius` works in ordinary CSS, it would be useful to cap the taxi radius based on the available space.  For instance, a div that's a 10px by 10px square can't have a radius greater than 5px effectively.

For the edges, the radius could be capped by the available space on the line.

If you have a larger radius, then the edge will bug out depending on the orientation of the nodes:


https://github.com/cytoscape/cytoscape.js/assets/989043/4fa3d97b-83fc-4b51-99a9-a082c33d2a01

Reply:
N.b. this issue also effects smaller radii, but not as obviously.